### PR TITLE
Transparent error code propagation in CAN MCAN module

### PR DESCRIPTION
**Pull request by KozhinovAlexander**
The implementation of can_mcan_start() function hides retun code of underlying function, which makes root-cause search mure difficult.
This change strives fro transparent error code propagation to higher software layers.

### CODE REVIEW

**Review by martinjaeger — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

File: drivers/can/can_mcan.c [hunk 1]
```diff
@@ -292,7 +292,7 @@ int can_mcan_start(const struct device *dev)
 {
 	const struct can_mcan_config *config = dev->config;
 	struct can_mcan_data *data = dev->data;
-	int err;
+	int err = 0;
```
Comment (jilaypandya) [inline drivers/can/can_mcan.c:295]: Just for my understanding.

Is this change really required? I don't see any code path where err would be read before getting assigned to some value.

Comment (martinjaeger) [reply drivers/can/can_mcan.c:295]: I was thinking the same. But I thought it would be OK to keep it for clarity reasons. We are returning `err` at the end now, so one one does not have to check every path in the code whether `err` is guaranteed to have a valid value.

**Review by jilaypandya — APPROVED** [review]
